go-controller/pkg/cni/ovs.go

```diff
@@ -260,23 +227,28 @@ func checkCancelSandbox(mac string, getter PodInfoGetter, namespace, name, nadNa
 	return nil
 }
 
-func waitForPodInterface(ctx context.Context, ifInfo *PodInterfaceInfo,
-	ifaceName, ifaceID string, getter PodInfoGetter,
+func waitForPodInterface(vsClient client.Client, ctx context.Context,
+	ifInfo *PodInterfaceInfo, ifaceName, ifaceID string, getter PodInfoGetter,
 	namespace, name, initialPodUID string) error {
 	var detail string
 	var ofPort int
-	var err error
+
+	waitTime := 20 * time.Millisecond
 
 	// DPUHost mode can't use OVS external IDs for port-up detection because
 	// there is no ovn-controller running in DPUHost mode to set port-up
 	checkExternalIDs := !ifInfo.IsDPUHostMode
 	if checkExternalIDs {
 		detail = " (ovn-installed)"
 	} else {
-		ofPort, err = getIfaceOFPort(ifaceName)
+		waitTime = 200 * time.Millisecond
+		ovsIface, err := libovsdbops.FindInterfaceByName(vsClient, ifaceName)
 		if err != nil {
 			return err
+		} else if ovsIface.Ofport == nil {
+			return fmt.Errorf("ovs interface %q ofport not set", ifaceName)
 		}
+		ofPort = *ovsIface.Ofport
```

Comment:
I got a bit confused as to why we would not be doing this in the `default` case below

Reply:
reworked this stuff in a fixup. I was following the original flow that read the ofport first, and then repeatedly during the loop, but it looks like maybe that's silly.